src/decoders/arw.rs

```diff
@@ -45,6 +45,14 @@ impl<'a> Decoder for ArwDecoder<'a> {
     };
     let src = &self.buffer[offset..];
 
+    // Lookup the orientation of the image to store for rotating after demosaic
```

Comment:
There's alot of duplication in tags. Most of the time the standard TIFF orientation tag should work. Probably all the time in case of Orientation as I think any format worth caring about implements it. dcraw as usual handles a bunch of the oddball ones but for this kind of thing they can just safely be ignored. Orientation is easy to fix manually if you actually do have one of the weirder formats.

As for the refactor I would try to make it so most formats don't even need any orientation code at all. check_supported() could probably stuff that info into the Camera struct and ok_image pick it up automatically. That way only formats that handle things more manually (like DNG or MRW) need to specifically call your orientation finding function.